src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -662,7 +662,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Return Nothing
                 End If
 
-                Return TextSpan.FromBounds(method.BlockStatement.Span.End, method.EndBlockStatement.SpanStart)
+                ' We don't want to include the BlockStatement or any trailing trivia up to and including its statement
+                ' terminator in the span. Instead, we use the start of the first statement's leading trivia (if any) up
+                ' to the start of the EndBlockStatement. If there aren't any statements in the block, we use the start
+                ' of the EndBlockStatements leading trivia.
+
+                Dim firstStatement = method.Statements.FirstOrDefault()
+                Dim spanStart = If(firstStatement IsNot Nothing,
+                                   firstStatement.FullSpan.Start,
+                                   method.EndBlockStatement.FullSpan.Start)
+
+                Return TextSpan.FromBounds(spanStart, method.EndBlockStatement.SpanStart)
```

Comment:
Can we add some XML doc comments to this method and in other places for what this span actually means? I'm guessing it answers the question of "where can I speculatively bind within this method", but it's not exactly clear.

Put another way: if I had to fix a bug in this method, how would I have any clue what it's expected to return, other than "it should return whatever it takes to get my tests passing?"